kubernetes-examples/src/main/java/io/fabric8/openshift/examples/BuildConfigExamples.java

```diff
@@ -37,106 +37,107 @@
-      log("Created namespace", client.namespaces().create(ns));
-
-      ServiceAccount fabric8 = new ServiceAccountBuilder().withNewMetadata().withName("fabric8").endMetadata().build();
-
-      client.serviceAccounts().inNamespace("thisisatest").create(fabric8);
-
-      log("Created image stream", client.imageStreams().inNamespace("thisisatest").createNew()
-        .withNewMetadata()
-        .withName("example-camel-cdi")
-        .endMetadata()
-        .withNewSpec()
-        .addNewTag()
-        .withName("latest")
-        .endTag()
-        .withDockerImageRepository("fabric8/example-camel-cdi")
-        .endSpec()
-        .withNewStatus().withDockerImageRepository("").endStatus()
-        .done());
-
-      log("Created image stream", client.imageStreams().inNamespace("thisisatest").createNew()
-        .withNewMetadata()
-        .withName("java-sti")
-        .endMetadata()
-        .withNewSpec()
-        .addNewTag()
-        .withName("latest")
-        .endTag()
-        .withDockerImageRepository("fabric8/java-sti")
-        .endSpec()
-        .withNewStatus().withDockerImageRepository("").endStatus()
-        .done());
-
-      log("Created build config", client.buildConfigs().inNamespace("thisisatest").createNew()
-        .withNewMetadata()
-        .withName("custom-build-config")
-        .endMetadata()
-        .withNewSpec()
-        .withServiceAccount("fabric8")
-        .withNewSource()
-        .withType("Git")
-        .withNewGit()
-        .withUri("https://github.com/fabric8io/example-camel-cdi.git")
-        .endGit()
-        .endSource()
-        .withNewResources()
-          .addToLimits("mykey", new Quantity("10"))
-          .addToRequests("mykey", new Quantity("10"))
-        .endResources()
-        .withNewStrategy()
-        .withType("Source")
-        .withNewSourceStrategy()
-        .withNewFrom().withName("java-sti:latest").withKind("DockerImage").endFrom()
-        .endSourceStrategy()
-        .endStrategy()
-        .withNewOutput()
-        .withNewTo().withKind("DockerImage").withName("example-camel-cdi:latest").endTo()
-        .endOutput()
-        .addNewTrigger()
-        .withType("GitHub")
-        .withNewGithub()
-        .withSecret("secret101")
-        .endGithub()
-        .endTrigger()
-        .endSpec()
-        .done());
-
-      Build build = client.buildConfigs().inNamespace("thisisatest").withName("custom-build-config").instantiate(new BuildRequestBuilder()
-        .withNewMetadata().withName("custom-build-config").endMetadata()
-        .build());
-      log("Build:", build.getMetadata().getName());
-
-      client.buildConfigs().inNamespace("thisisatest").withName("custom-build-config")
-        .withSecret("secret101")
-        .withType("github")
-        .trigger(new WebHookTriggerBuilder()
+    try(KubernetesClient kubernetesClient = new DefaultKubernetesClient(config)){
+      OpenShiftClient client = kubernetesClient.adapt(OpenShiftClient.class);
+
+      try {
```

Comment:
It's the same for **every** example.
Please reuse the try-catch block and remove the finally clause (if applicable) where the client is closed.
```
try (KubernetesClient kubernetesClient = new DefaultKubernetesClient(config)) {.
  OpenShiftClient ifNecessary = kubernetesClient.adapt(OpenShiftClient.class);
  /* ... your code with no extra try-catch blocks (unless strictly necessary) ... */
}
```

Reply:
@manusa Sorry for not responding earlier, I have an exam today at 1 pm IST. I'll do the required changes after that exam gets over.

Reply:
:) take your time